### PR TITLE
Use generic resource handling

### PR DESCRIPTION
**Pull request by skitt**
Depends on https://github.com/submariner-io/admiral/pull/756
Depends on https://github.com/submariner-io/admiral/pull/760
Depends on https://github.com/submariner-io/cloud-prepare/pull/817
Depends on https://github.com/submariner-io/lighthouse/pull/1402
Depends on https://github.com/submariner-io/submariner/pull/2742
Depends on https://github.com/submariner-io/submariner-operator/pull/2839

### CODE REVIEW

File: pkg/serviceaccount/ensure.go [hunk 1]
```diff
@@ -45,8 +44,8 @@ const (
 
 func ensure(ctx context.Context, kubeClient kubernetes.Interface, namespace string, sa *corev1.ServiceAccount) (bool, error) {
 	result, err := util.CreateOrUpdate(ctx, resource.ForServiceAccount(kubeClient, namespace), sa,
```
Comment (tpantelis) [inline pkg/serviceaccount/ensure.go:46]:
```suggestion
	result, err := util.CreateOrUpdate[*corev1.ServiceAccount](ctx, resource.ForServiceAccount(kubeClient, namespace), sa,
```

Comment (skitt) [reply pkg/serviceaccount/ensure.go:46]: I rather like letting the compiler infer the type :wink:

Comment (tpantelis) [reply pkg/serviceaccount/ensure.go:46]: yeah but you've specified the type explicitly in other places so for consistency.... but it doesn't matter either way, that's why I didn't -1 😄

Comment (skitt) [reply pkg/serviceaccount/ensure.go:46]: Ah, I thought I’d dropped the type parameters everywhere they weren’t needed, but I missed some.

**Review by tpantelis — APPROVED** [review]

**Review by sridhargaddam — APPROVED** [review]

**Pull request merged**